Example/Emission/AppDelegate.m

```diff
@@ -70,7 +70,15 @@ - (BOOL)application:(UIApplication *)application didFinishLaunchingWithOptions:(
   if ([auth isAuthenticated]) {
     NSString *accessToken = [auth token];
     if (accessToken) {
-      [self setupEmissionWithUserID:[auth userID] accessToken:accessToken keychainService:service];
+
+      [auth validateStoredCredentialsPassing:^{
+        [self setupEmissionWithUserID:[auth userID] accessToken:accessToken keychainService:service];
+      } failing:^{
+        [auth presentAuthenticationPromptOnViewController:rootVC completion:^{
+          NSLog(@"Logged in successfully :)");
+          [self setupEmissionWithUserID:[auth userID] accessToken:[auth token] keychainService:service];
+        }];
```

Comment:
could be DRYer (its the same as line 84) - but it's not really enough to worry me at this scale